### PR TITLE
[KZL 380] Bulk controller

### PR DESCRIPTION
**Pull request by alexandrebouthinon**
## What does this PR do ?
Provide:
- Bulk controller unit testing
- Bulk controller refactoring

### CODE REVIEW

**Review by jenow — APPROVED** [review]

File: src/Bulk.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+     */
+    public function import(array $data, array $options = [])
+    {
+        $response = $this->kuzzle->query(
+            $this->kuzzle->buildQueryArgs('bulk', 'import'),
+            [
+                'body' => [
+                    'bulkData' => json_encode($data)
+                ]
+
```
Comment (Aschen) [inline src/Bulk.php]: Nitpicking:
Unnecessary blank line

File: composer.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "kuzzleio/kuzzle-sdk",
-  "version": "3.0.3",
+  "version": "3.0.4",
```
Comment (Aschen) [inline composer.json]: Why only minor update ?

File: src/Kuzzle.php [hunk 3]
```diff
@@ -116,7 +122,10 @@ public function __construct($host, array $options = [])
         $this->url = 'http://' . $host . ':' . $this->port;
```
Comment (Aschen) [inline src/Kuzzle.php:123]: What if Kuzzle is accessible only in HTTPS ?

File: composer.json (hunk 2 above)
Comment (alexandrebouthinon) [reply composer.json]: I don't know why but I did not change this field.

**Review by Aschen — APPROVED** [review]

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+-->
+
+<!--
+  IMPORTANT
+  Don't forget to add the corresponding "changelog:xxx" label to your PR.
+  This is part of our release process in order to generate the change log.
+-->
+
+
+## What does this PR do ?
```
Comment (scottinet) [inline .github/pull_request_template.md]: no space before question marks

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+<!--
+  This template is not mandatory.
```
Comment (scottinet) [inline .github/pull_request_template.md]: nitpicking: `This template is optional.`

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  It simply serves to provide a guide to allow a better review of pull requests.
```
Comment (scottinet) [inline .github/pull_request_template.md]: `It simply servers as a guide...`

File: src/config/routes.json [hunk 7]
```diff
@@ -59,7 +59,7 @@
   "bulk": {
     "import": {
       "name": "import",
-      "route": "/:index/:collection/_bulk",
+      "route": "/bulk/import",
```
Comment (scottinet) [inline src/config/routes.json]: it's one of those: 

```
  {verb: 'post', url: '/_bulk', controller: 'bulk', action: 'import'},
  {verb: 'post', url: '/:index/_bulk', controller: 'bulk', action: 'import'},
  {verb: 'post', url: '/:index/:collection/_bulk', controller: 'bulk', action: 'import'},
```

**Review by scottinet — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
+<!--
+  This template is optional.
+  It simply servers as a guide to allow a better review of pull requests.
```
Comment (jenow) [inline .github/pull_request_template.md]: I think @scottinet meant "it simply serves as a guide"

**Review by benoitvidis — APPROVED** [review]

File: .github/pull_request_template.md (hunk 8 above)
Comment (alexandrebouthinon) [reply .github/pull_request_template.md]: Ôoops

**Review by scottinet — APPROVED** [review]

**Pull request merged**